contracts/external/oracles/ChainlinkAutomationPriceOracle.sol

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+/*
+
+    Copyright 2023 Dolomite
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+*/
+
+pragma solidity ^0.8.9;
+
+import { IChainlinkAutomationPriceOracle } from "../interfaces/IChainlinkAutomationPriceOracle.sol";
+import { OnlyDolomiteMargin } from "../helpers/OnlyDolomiteMargin.sol";
+
+import { Require } from "../../protocol/lib/Require.sol";
+
+
+/**
+ * @title   ChainlinkAutomationPriceOracle
+ * @author  Dolomite
+ *
+ * @notice  An abstract contract that implements the IChainlinkAutomationPriceOracle interface
+ * @notice  Contains variables and functions for Chainlink Automation
+ */
+abstract contract ChainlinkAutomationPriceOracle is IChainlinkAutomationPriceOracle, OnlyDolomiteMargin {
+
+    // ============================ Constants ============================
+
+    bytes32 private constant _FILE = "ChainlinkAutomationPriceOracle";
+    uint256 private constant _ONE_UNIT = 10 ** 18;
+
+    // ============================ Public State Variables ============================
+
+    uint256 public HEARTBEAT = 12 * 3600; // solhint-disable-line var-name-mixedcase
+    uint256 public GRACE_PERIOD = 3600; // solhint-disable-line var-name-mixedcase
+    uint256 public UPPER_EDGE = 10025; // solhint-disable-line var-name-mixedcase
+    uint256 public LOWER_EDGE = 9975; // solhint-disable-line var-name-mixedcase
+    address public CHAINLINK_REGISTRY; // solhint-disable-line var-name-mixedcase
```

Comment:
Set these in the constructor with calls to `_ownerSet...` so the events fire and the way the variables are set is standardized

Reply:
Do you want me to adjust the constructor to accept these arguments or simply use those default values within the _ownerSet...